extensions/http/src/main/java/org/mule/extension/http/internal/request/validator/HttpRequesterProvider.java

```diff
@@ -215,6 +204,30 @@ private void verifyConnectionsParameters() throws InitialisationException {
     }
   }
 
+  @Override
+  public HttpClient connect() throws ConnectionException {
+    String threadNamePrefix = String.format(THREAD_NAME_PREFIX_PATTERN, ThreadNameHelper.getPrefix(muleContext), configName);
+
+    HttpClientConfiguration configuration = new HttpClientConfiguration.Builder()
+        .setUriParameters(new DefaultUriParameters(protocol, host, port)).setAuthentication(authentication)
+        .setTlsContextFactory(tlsContextFactory).setProxyConfig(proxyConfig).setClientSocketProperties(clientSocketProperties)
+        .setMaxConnections(maxConnections).setUsePersistentConnections(usePersistentConnections)
+        .setConnectionIdleTimeout(connectionIdleTimeout).setThreadNamePrefix(threadNamePrefix).setOwnerName(configName).build();
+
+    HttpClientFactory httpClientFactory = muleContext.getRegistry().get(OBJECT_HTTP_CLIENT_FACTORY);
```

Comment:
it doesn't matter what it's a part of. You shouldn't expect the registry to return different instances for the same key. So, inject it instead of doing a lookup